java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/MultiCryptoProviderDecryptionTest.java

```diff
@@ -0,0 +1,309 @@
+
+    }
+
+    @Test
+    private void decryptAES256CBCPKCS5Padding() throws IOException {
+
+        final SupportedCipherDetails cipherDetails = CIPHERS_TO_TEST.get(8);
+        validateEncryptedFileWithHeaderValues(cipherDetails);
+
+    }
+
+    /* Test Utility methods used in all unit-tests. */
+
+    private void validateEncryptedFileWithHeaderValues(SupportedCipherDetails cipherDetails) throws IOException {
+
+        final String cipherName = cipherDetails.getCipherId().replaceAll("/","-");
+        final ConfigContext config = new ChainedConfigContext(
+                new DefaultsConfigContext(),
+                new EnvVarConfigContext(),
+                new SystemSettingsConfigContext())
+
+                /* Encryption settings */
+                .setClientEncryptionEnabled(true)
+                .setEncryptionAlgorithm(cipherDetails.getCipherId())
+                .setEncryptionKeyId("personal-256")
+                .setEncryptionPrivateKeyBytes(SECRET_KEY)
+                .setEncryptionAuthenticationMode(EncryptionAuthenticationMode.Mandatory);
+
+
+        client = new MantaClient(config);
+
+        final SecretKey secretKey = generateSecretKey(cipherDetails);
+        final EncryptionContext ctx = new EncryptionContext(secretKey, cipherDetails, true);
+
+        final String filePath = "test-data/ciphertext/" + cipherName + ".ciphertext";
+        final URL encryptedFileURL = getClass().getClassLoader().getResource(filePath);
+
+        MantaHttpHeaders getResponseHttpHeaders = new MantaHttpHeaders();
+        populateHeaderValuesFromResources(getResponseHttpHeaders,cipherDetails);
+
+
+        final long cipherTextSize = getResponseHttpHeaders.getContentLength();
+
+        try (InputStream encryptedStream = encryptedFileURL.openStream();
+             BoundedInputStream bin = new BoundedInputStream(encryptedStream, cipherTextSize)) {
+
+            final byte[] actualBytes = IOUtils.toByteArray(bin);
+
+            try (final MantaEncryptedObjectInputStream decryptingStream = new MantaEncryptedObjectInputStream(
+                    new MantaObjectInputStream(
+                            new MantaObjectResponse("/path", getResponseHttpHeaders),
+                            Mockito.mock(CloseableHttpResponse.class),
+                            new EofSensorInputStream(
+                                    new ByteArrayInputStream(actualBytes),
+                                    null)),
+                    cipherDetails,
+                    secretKey,
+                    true)) {
+
+                ByteArrayOutputStream decrypted = new ByteArrayOutputStream();
+                IOUtils.copy(decryptingStream, decrypted);
+                decryptingStream.close();
+                assertByteArrayEquals(expectedContent, decrypted.toByteArray());
+            }
+        }
+    }
+
+    private Map<String, Object> parseHeaderFileInfoStoreInMap(final URL headerURL) throws  IOException {
+        Scanner scanner = null;
```

Comment:
You never use the assigned null value here. Why not do: `final Scanner scanner;`?